src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -44,7 +47,9 @@ export default function (chrome, internals) {
         onRouteChange();
 
         // and some local values
-        chrome.httpActive = $http.pendingRequests;
+        $scope.$watch(() => $http.pendingRequests, (pendingHttpRequests) => {
+          chrome.httpActive = pendingHttpRequests.filter(request => !systemApi.isSystemApiRequest(request));
+        }, true);
```

Comment:
@ycombinator I think so, yes. That's what `$watchCollection` is meant for.

> Shallow watches the properties of an object and fires whenever any of the properties change (for arrays, this implies watching the array items; for object maps, this implies watching the properties). If a change is detected, the listener callback is fired.

[Source](https://docs.angularjs.org/api/ng/type/$rootScope.Scope)